src/shainet/network.cr

```diff
@@ -513,6 +514,60 @@ module SHAInet
       @all_neurons.each &.randomize_bias
     end
 
+    def save_to_file(file_path : String)
+      dump_network = Array(Hash(String, String | Array(Hash(String, Array(Hash(String, String | Float64)) | Float64 | String | String)))).new
+
+      [@input_layers, @output_layers, @hidden_layers].flatten.each do |layer|
+        dump_layer = Hash(String, String | Array(Hash(String, Array(Hash(String, String | Float64)) | Float64 | String | String))).new
+        dump_neurons = Array(Hash(String, Array(Hash(String, String | Float64)) | Float64 | String | String)).new
+        layer.neurons.each do |neuron|
+          n = Hash(String, Array(Hash(String, String | Float64)) | Float64 | String | String).new
+          n["id"] = neuron.id
+          n["activation"] = neuron.activation
+          n["n_type"] = neuron.n_type.to_s
+          n["synapses_in"] = Array(Hash(String, String | Float64)).new
+          n["synapses_out"] = Array(Hash(String, String | Float64)).new
+          neuron.synapses_in.each do |s|
+            s_h = Hash(String, String | Float64).new
+            s_h["source"] = s.source_neuron.id
+            s_h["destination"] = s.dest_neuron.id
+            s_h["weight"] = s.weight
+            n["synapses_in"].as(Array(Hash(String, String | Float64))) << s_h
+          end
+          neuron.synapses_out.each do |s|
+            s_h = Hash(String, String | Float64).new
+            s_h["source"] = s.source_neuron.id
+            s_h["destination"] = s.dest_neuron.id
+            s_h["weight"] = s.weight
+            n["synapses_out"].as(Array(Hash(String, String | Float64))) << s_h
+          end
+          dump_neurons << n
+        end
+
+        l_type = ""
+        if @input_layers.includes?(layer)
+          l_type = "input"
```

Comment:
what do you think of moving the `type` property in the layer itself instead of having 3 arrays to identify them?

Reply:
I thought about it, basically in the end we will move one "multi check" to another part of the network that needs to know which is which.